### DISCUSSION TITLE
Kesalahan pada pembuatan time series? (Capstone ML)

### DISCUSSION THREAD

**Discussion by AGelfP** [teamalgoritma/community#628] [Q&A] [answered]
Data yang digunakan adalah data kasus Time Series FnB

Sepertinya saya masih ada kesalahan dalam menggunakan fungsi ts(). Berdasarkan informasi yang disediakan pada kasus, toko dibuka setiap hari, selama 12 jam (10 sampai 22). Berdasarkan dokumentasi fungsi ts() yang saya temukan, parameter frequency bertujuan untuk "memberi tahu" object kita bahwa jumlah observasi yang ada per unit waktu yang terdapat pada waktu kita. Karena object yang ingin dibuat adalah untuk data 12 jam per hari, saya memberikan nilai 12 pada parameter frequency

```
train_ts<- ts(
train_cln2$visitor, # train_cln adalah data yang sudah saya filter untuk hanya include business hours, $visitor adalah jumlah pengunjung yang datang setiap jamnya
   frequency = 12 # 12 karena data untuk setiap harinya, 12 jam per hari
)

```

Namun ketika saya visualisasikan dengan autoplot(), sumbu x tidak menunjukkan waktu dalam bentuk format yang diharapkan. Format seharusnya YYYY-MM-DD HH:MM:SS, namun hasilnya malah muncul seperti ini

![image](https://user-images.githubusercontent.com/122977451/231656822-ebdf3ff4-4601-4921-b223-6f31543f6733.png)

Apakah ini indikasi bahwa ada langkah yang kurang tepat? Atau apakah saya salah mengerti parameter frequency?

**Comment by kinantytasyao** [comment, marked as answer]
Dear @AGelfP 

Langkah yang Bapak lakukan sudah tepat, sumbu axis x pada autoplot sekarang menunjukkan label satuan jam. Yang perlu Bapak periksa kembali adalah frekuensi yang digunakan, apakah benar dari jam 10 pagi sampai jam 10 malam adalah 12 jam.

Semoga jawaban ini dapat membantu.
Terima kasih

**Reply by AGelfP** [reply]
Selamat sore Kak, artinya jam 10 sebagai jam buka toko juga diinclude menjadi observasi dong ya? Frequency sudah saya ganti jadi 13 untuk menginclude jam buka tersebut. Tapi di x-axisnya masih tidak menunjukkan timestamp Kak.

![image](https://user-images.githubusercontent.com/122977451/231715612-1f45c036-e672-42a6-897c-2379f0893689.png)

Apakah artinya masih terdapat kesalahan di frequency atau emang outputnya seperti gambar tertera ya Kak? Terima kasih

**Reply by kinantytasyao** [reply]
Benar Pak, jam 10 nya diinclude menjadi observasi. Untuk autoplot sudah benar dan tidak ada masalah Pak, label axis x di atas menunjukkan jam secara compound.

**Reply by AGelfP** [reply]
baik Kak kalau begitu, terima kasih yaa

**Reply by kinantytasyao** [reply]
Sama sama Pak Gelfand
